Spanner/src/Session/CacheSessionPool.php

```diff
@@ -313,6 +314,9 @@ public function acquire($context = SessionPoolInterface::CONTEXT_READ)
 
         // If we don't have a session, let's wait for one or throw an exception.
         if (!$session) {
+            if (!is_null($exception)) {
+                throw $exception;
```

Comment:
There is a potential issue here. The function as-is throws a `\RuntimeException`, but the `createSessions` function catches all exceptions of the base class `\Exception`. This means potentially someone's existing code could break if they're catching `\RuntimeException` but not `\Exception`, and we throw an exception of `\Exception`. instead, I would suggest doing something like this:

```suggestion
                throw $exception instanceof RuntimeException
                    ? $exception
                    : new RuntimeException($exception->getMessage(), $exception->getCode(), $exception);
```